emcc.py

```diff
@@ -2165,6 +2163,23 @@ def check_memory_setting(setting):
   if settings.SINGLE_FILE:
     settings.GENERATE_SOURCE_MAP = 0
 
+  if settings.EVAL_CTORS:
+    if settings.WASM2JS:
+      # code size/memory and correctness issues TODO
+      diagnostics.warning('emcc', 'disabling EVAL_CTORS due to wasm2js')
+      settings.EVAL_CTORS = 0
+    elif settings.USE_PTHREADS:
+      diagnostics.warning('emcc', 'disabling EVAL_CTORS due to pthreads (passive segments)')
+      settings.EVAL_CTORS = 0
+    elif settings.RELOCATABLE:
+      diagnostics.warning('emcc', 'disabling EVAL_CTORS due to relocatable (movable segments)')
+      settings.EVAL_CTORS = 0
+    elif settings.ASYNCIFY:
+      # In Asyncify exports can be called more than once, and this seems to not
+      # work properly yet (see test_emscripten_scan_registers).
+      diagnostics.warning('emcc', 'disabling EVAL_CTORS due to asyncify')
```

Comment:
I all these cases I think its better to just error our given contradictory settings.  That is what we normally do.